client/src/elm/Pages/NCD/RecurrentActivity/Update.elm

```diff
@@ -0,0 +1,564 @@
+
+                updatedData =
+                    model.labResultsData
+                        |> (\data -> { data | urineDipstickTestForm = updatedForm })
+            in
+            ( { model | labResultsData = updatedData }
+            , Cmd.none
+            , []
+            )
+
+        SaveUrineDipstickResult personId saved nextTask ->
+            let
+                measurementId =
+                    Maybe.map Tuple.first saved
+
+                measurement =
+                    getMeasurementValueFunc saved
+
+                extraMsgs =
+                    generateLabResultsMsgs nextTask
+
+                appMsgs =
+                    toUrineDipstickResultsValueWithDefault measurement model.labResultsData.urineDipstickTestForm
+                        |> Maybe.map
+                            (Backend.NCDEncounter.Model.SaveUrineDipstickTest personId measurementId
+                                >> Backend.Model.MsgNCDEncounter id
+                                >> App.Model.MsgIndexedDb
+                                >> List.singleton
+                            )
+                        |> Maybe.withDefault []
+            in
+            ( model
+            , Cmd.none
+            , appMsgs
+            )
+                |> sequenceExtra (update language currentDate id db) extraMsgs
+
+        SetRandomBloodSugar value ->
+            let
+                form =
+                    model.labResultsData.randomBloodSugarTestForm
+
+                updatedForm =
+                    { form | sugarCount = String.toFloat value }
+
+                updatedData =
+                    model.labResultsData
+                        |> (\data -> { data | randomBloodSugarTestForm = updatedForm })
+            in
+            ( { model | labResultsData = updatedData }
+            , Cmd.none
+            , []
+            )
+
+        SaveRandomBloodSugarResult personId saved nextTask ->
+            let
+                measurementId =
+                    Maybe.map Tuple.first saved
+
+                measurement =
+                    getMeasurementValueFunc saved
+
+                extraMsgs =
+                    generateLabResultsMsgs nextTask
+
+                appMsgs =
+                    toRandomBloodSugarResultsValueWithDefault measurement model.labResultsData.randomBloodSugarTestForm
+                        |> Maybe.map
+                            (Backend.NCDEncounter.Model.SaveRandomBloodSugarTest personId measurementId
+                                >> Backend.Model.MsgNCDEncounter id
+                                >> App.Model.MsgIndexedDb
+                                >> List.singleton
+                            )
+                        |> Maybe.withDefault []
+            in
+            ( model
+            , Cmd.none
+            , appMsgs
+            )
+                |> sequenceExtra (update language currentDate id db) extraMsgs
+
+        -- SetActiveNextStepsTask task ->
```

Comment:
Remove or `@todo`

Reply:
This all has been implemented at #456 and #459 already.